From b8abbe4fc8c67f5f5a182f6a756fcdfc0145e487 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Tue, 26 Feb 2019 17:09:15 +0100
Subject: [PATCH] nand_base: Fix prototype of spinand_flash_detect()

This function returns a pointer to struct nand_flash_dev and not an int, use the correct type.

This fixes a compile warning.
---
 drivers/mtd/nand/nand_base.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/nand/nand_base.c b/drivers/mtd/nand/nand_base.c
index 0e55c45bcb9a..c139838b3722 100644
--- a/drivers/mtd/nand/nand_base.c
+++ b/drivers/mtd/nand/nand_base.c
@@ -49,7 +49,8 @@
 #include <linux/mtd/nand_ecc_on_die.h>
 
 #ifdef CONFIG_MTD_LTQ_SPINAND
-int spinand_flash_detect(struct mtd_info *mtd, struct nand_chip *chip);
+struct nand_flash_dev *spinand_flash_detect(struct mtd_info *mtd,
+					    struct nand_chip *chip);
 #endif /* CONFIG_MTD_LTQ_SPINAND */
 
 static int nand_get_device(struct mtd_info *mtd, int new_state);
